### PR TITLE
Subsample dataset by volume fix

### PR DESCRIPTION
**Pull request by z-fabian**
If `sample_rate < 1.0`, `SliceDataset` subsamples the training dataset by slices and not by volumes. This is undesired since 
1. Based on the function documentation says `sample_rate` should control what fraction of the volumes should be loaded. 
2. Subsampling by volumes makes more sense in practice. Randomly selected slices can create very unbalanced datasets (possibly lots of empty slices) and subsampling by volumes mimics the real life case of having less MRI volumes for training.  

This fix modifies `SliceDataset` to subsample by volumes when `sample_rate < 1.0`. If subsampling by slice is the desired behavior, then we should update the fuction helper to reflect that.

### CODE REVIEW

**Review by mmuckley — CHANGES_REQUESTED** [review]
My worry here is we're breaking backwards compatibility. If someone has a project that depends on the repository and they want to reproduce an old result, this may break their code. Instead I would recommend a new variable, such as `volume_sample_rate`, and make sure that `sample_rate` and `volume_sample_rate` aren't set simultaneously, throwing an exception if they are. This might be best if both defaulted to `None` rather than `1.0`, but that's open to the implementation as you wish..

**Comment by z-fabian** [comment]
@mmuckley As you suggested I added a new argument `volume_sample_rate`. If it is not set, `sample_rate` will behave the same way as before. If it is set, but `sample_rate` is not, then we subsample by volumes. I changed the default values for both of these arguments to None, in which case `1.0` will be used. Should be 100% backwards compatible.

File: fastmri/data/mri_data.py [hunk 1]
```diff
@@ -123,20 +124,40 @@ def __init__(
...
                 useful for large datasets like the brain data.
             dataset_cache_file: Optional; A file in which to cache dataset
                 information for faster load times.
             num_cols: Optional; If provided, only slices with the desired
                 number of columns will be considered.
         """
+        if sample_rates is not None and volume_sample_rates is not None:
+            raise ValueError(
+                "either set sample_rates (sample by slices) or volume_sample_rates (sample by volumes) but not both"
+            )
```
Comment (mmuckley) [inline fastmri/data/mri_data.py:147]: :+1:

File: fastmri/data/mri_data.py [hunk 2]
```diff
@@ -205,6 +234,11 @@ def __init__(
         if challenge not in ("singlecoil", "multicoil"):
             raise ValueError('challenge should be either "singlecoil" or "multicoil"')
 
+        if sample_rate is not None and volume_sample_rate is not None:
+            raise ValueError(
+                "either set sample_rate (sample by slices) or volume_sample_rate (sample by volumes) but not both"
+            )
```
Comment (mmuckley) [inline fastmri/data/mri_data.py:240]: 👍🏻

File: fastmri/pl_modules/data_module.py [hunk 3]
```diff
@@ -185,10 +206,12 @@ def prepare_data(self):
                 zip(data_paths, data_transforms)
             ):
                 sample_rate = self.sample_rate if i == 0 else 1.0
+                volume_sample_rate = self.volume_sample_rate if i == 0 else None
```
Comment (mmuckley) [inline fastmri/pl_modules/data_module.py:209]: Funny enough, I think this is unnecessary.

**Review by mmuckley — APPROVED** [review]
Looks good to me! Great work!

**Pull request merged**